api/src/main/java/javax/enterprise/context/spi/ManageableContext.java

```diff
@@ -0,0 +1,116 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.context.spi;
+
+import javax.enterprise.context.ContextNotActiveException;
+import java.lang.annotation.Annotation;
+
+/**
+ * Provides the same capabilities as a {@linkplain javax.enterprise.context.spi.Context object}, however is not intended to be active when first retrieved.
+ * ManageableContext is designed as a tool to allow you to start and stop a built in context programmatically in ways other than how the spec defines them.
+ *
+ * <pre>
+ *
+ * public class MyRunner implements Runnable {
+ *
+ *     public void run() {
+ *        BeanManager beanManager = CDI.current().getBeanManager();
+ *        ManageableContext context = beanManager.getManageableContext(RequestScoped.class);
+ *        context.activate();
+ *        // do some work in this runnable
+ *        context.deactivate();
+ *     }
+ *
+ * }
+
+ * </pre>
+ *
+ * @author John D. Ament
+ * @since 2.0
+ */
+public interface ManageableContext extends AutoCloseable{
+
+    /**
+     * Get the scope type of the context object.
+     *
+     * @return the scope
+     */
+    Class<? extends Annotation> getScope();
+
+    /**
+     * Return an existing instance of certain contextual type or create a new instance by calling
+     * {@link javax.enterprise.context.spi.Contextual#create(CreationalContext)} and return the new instance.
+     *
+     * @param <T> the type of contextual type
+     * @param contextual the contextual type
+     * @param creationalContext the context in which the new instance will be created
+     * @return the contextual instance
+     *
+     * @throws ContextNotActiveException if the context is not active
+     */
+    <T> T get(Contextual<T> contextual, CreationalContext<T> creationalContext);
+
+    /**
+     * Return an existing instance of a certain contextual type or a null value.
+     *
+     * @param <T> the type of the contextual type
+     * @param contextual the contextual type
+     * @return the contextual instance, or a null value
+     *
+     * @throws ContextNotActiveException if the context is not active
+     */
+    <T> T get(Contextual<T> contextual);
+
+    /**
+     * Determines if the context object is active.
+     *
+     * @return <tt>true</tt> if the context is active, or <tt>false</tt> otherwise.
+     */
+    boolean isActive();
+
+    /**
+     * <p>
+     * Destroy the existing contextual instance. If there is no existing instance, no action is taken.
+     * </p>
+     *
+     * @param contextual the contextual type
+     * @throws ContextNotActiveException if the context is not active
+     */
+    void destroy(Contextual<?> contextual);
+
+    /**
+     * Starts the given context, allowing beans to be looked up within the bounds of that context.
+     *
+     * @throws IllegalStateException if the context has already been started.
+     */
+    void activate();
```

Comment:
This method could return the context instance so that it's easier to use with `try-with-resource` block:
````
try (ManageableContext ctx = beanManager.createManagedContext(RequestScoped.class).activate()) {
  ...
}
````